src/sagemaker_containers/training.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import importlib
+import subprocess
+
+import sys
+
+import sagemaker_containers.training_environment as env
+
+
+class Training(object):
+    def __init__(self, train_fn=None, module_name=None):
+        self.environment = env.TrainingEnvironment()
+        self.train_fn = train_fn
+        self.module_name = module_name
+
+    @classmethod
+    def from_train_fn(cls, train_fn):
+        return cls(train_fn=train_fn)
+
+    @classmethod
+    def from_module(cls, module_name):
+        return cls(module_name=module_name)
+
+    def start(self):
+
+        try:
+            if self.train_fn:
+
+                user_module = self.import_user_module(self.environment.code_dir)
+
+                self.train_fn(user_module, self.environment)
+            else:
+                subprocess.Popen(['python', '-m', self.module_name])
```

Comment:
this asumes that the module is importing this as a library in order to figure out where to put stuff right? or the user just hardcoded all their paths but we don't care how they do it as long as it works right?

Reply:
When the user provides a module, we will validate it before starting the training job.
The module will be saved to the code_dir. When a new Environment instance is created, it adds the code dir to PYTHONPATH.